### ISSUE TITLE
Using dbt-duckdb with meltano, external sources not read correctly? 

### ISSUE THREAD

**Issue by andres-sellforte** [duckdb/dbt-duckdb#481]
Hi all, I'm not sure if this is the correct place to post my question, but I have been struggling with the following for a while:

I have a meltano enviroment where I pull data from a tap and use the target-parquet to save it as parquet files, all well there. Then I want to bring dbt into the process, and then I create my sources file:

```yml
sources:
  - name: awin-local
    meta:
      external_location: "{{ env_var('OUTPUT_PARQUET', 'data') }}/{name}-20241126_145148-0-0.gz.parquet"
    tables:
      - name: transactions
```

the OUTPUT_PARQUET is coming from the `meltano.yml `so I expect this external source to point to my parquet file, I want to run queries against using dbt

Now I try a model/all_data.sql

```sql
with trns as (
    select networkFee__amount from {{ source('awin-local', 'transactions') }}
)

select * from trns
```
and then I compile ( `uvx meltano invoke dbt-duckdb compile` ) to see the SQL and I get the following query

```sql
with trns as (
    select networkFee__amount from "main"."awin-local"."transactions"
)

select * from trns
```

and this fails to run ( ` dbt-duckdb run`), what I'm expecting to see is something more like

```sql
with trns as (
    select networkFee__amount from "output/awin-parquet/transactions/{name}-20241126_145148-0-0.gz.parquet"
)
select * from trns
```

Based on my definition of the source and so that I can run queries against the parquet file. But I can't, and I've been unable to figure it out, any ideas? Thank you very much for your help

**Comment by edgarrmondragon** [comment]
Slack thread: https://meltano.slack.com/archives/C069RGRJYD7/p1732692405547239

**Comment by jwills** [comment]
hey @edgarrmondragon I'm traveling around for the next few days-- did this get resolved or no?

**Comment by edgarrmondragon** [comment]
> hey @edgarrmondragon I'm traveling around for the next few days-- did this get resolved or no?

Not yet, I've yet to set up a local MRE. Feel free to assign me to this issue.

**Comment by edgarrmondragon** [comment]
@andres-sellforte

TL;DR I wasn't able to reproduce this but I got a repo[^1] with a similar project using target-parquet and dbt-duckdb.

A few things that may differ and you might wanna double check:

1. I used a fork of target-parquet that I created a while ago to support Python 3.13
2. I rely on the built-in `MELTANO_PROJECT_ROOT` env var, instead of `OUTPUT_PARQUET`

[^1]: https://github.com/edgarrmondragon/meltano-parquet-dbt-duckdb/

**Comment by andres-sellforte** [comment]
Hi! 

Thanks for looking into it, your repo helped me, and I was able to fix the sources problem in my project

 The main difference between seems to be that my original meltano.yml had:

```yml
  transformers:
    - name: dbt-duckdb
      variant: jwills
      pip_url: dbt-core~=1.2.0 dbt-duckdb~=1.2.0
```

but in your repo those are listed as utilities: 

```yml
  utilities:
    - name: dbt-duckdb
      variant: jwills
      pip_url: dbt-core dbt-duckdb meltano-dbt-ext~=0.3.0
```
not very sure why the difference. I think i just used `meltano add` to add those, so I'm unclear 

after switching, the `dbt-duckdb:compile` generated the correct query. 

Thanks!
